src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -87,4 +89,9 @@ public VectorQueryContext vectorContext()
             vectorContext = new VectorQueryContext(readCommand);
         return vectorContext;
     }
+    
+    public boolean useStrictFiltering()
+    {
+        return readCommand.rowFilter().isStrict() || !hasUnrepairedMatches;
```

Comment:
We arrive here w/ a non-strict row filter and no unrepaired matches in reads in `PartialUpdateHandlingTest` that happen immediately after repair (in `validateCurrent()`).